### PR TITLE
APPENG-3377: Implement evaluate analysis tool

### PR DESCRIPTION
**Pull request by YosiElias**
- Implements the evaluate_analysis tool that reviews LLM analysis results for non-final issues and
determines finality using IssueAnalysisService._recommend. 
- Includes comprehensive unit tests covering core
functionality, error handling, and edge cases.

### CODE REVIEW

**Comment by GuyZivRH** [comment]
lgtm

File: src/sast_agent_workflow/tools/evaluate_analysis.py [hunk 1]
```diff
@@ -2,44 +2,148 @@
...
+        issue_id: The issue identifier
+        issue_data: The PerIssueData object
+        issue_analysis_service: The IssueAnalysisService instance
+        llm: The LLM to use for recommendations
+    """
+    if not issue_id or not isinstance(issue_id, str):
+        logger.error("Invalid issue_id: must be a non-empty string")
+        return
+    
+    if not isinstance(issue_data, PerIssueData):
```
[the remark this answers is not in the document]
Comment (YosiElias) [reply src/sast_agent_workflow/tools/evaluate_analysis.py:35]: You're right that tracker is a Pydantic model, but Pydantic validates during model creation, and the isinstance check protects against post-creation data corruption and manual dict manipulation. Regarding the type signature, it serves IDE purposes but isn't runtime-enforced.

File: src/sast_agent_workflow/tools/evaluate_analysis.py [hunk 2]
```diff
@@ -2,44 +2,148 @@
...
+from dto.SASTWorkflowModels import SASTWorkflowTracker, PerIssueData
+from dto.LLMResponse import AnalysisResponse, FinalStatus
+from dto.ResponseStructures import JudgeLLMResponse, RecommendationsResponse
+from services.issue_analysis_service import IssueAnalysisService
+from services.vector_store_service import VectorStoreService
 
 logger = logging.getLogger(__name__)
 
 
+def _process_single_issue_evaluation(issue_id: str, issue_data: PerIssueData,
```
[the remark this answers is not in the document]
Comment (YosiElias) [reply src/sast_agent_workflow/tools/evaluate_analysis.py]: [Move helper function to bottom of the file](https://github.com/RHEcosystemAppEng/sast-ai-workflow/pull/101/commits/4d953a1581eec606864df5d99c13eb9c8cba24ca)
